app/controllers/admin/books_controller.rb

```diff
@@ -1,5 +1,6 @@
 class Admin::BooksController < Admin::BaseController
   before_action :find_book, except: :index
+  load_and_authorize_resource
```

Comment:
```suggestion
  authorize_resource
```
chị đổi thành ri có được ko, và giải thích sự khác nhau nhé

Reply:
Trong trường hợp này vì em đã có sử dụng before_action find_book, nên có thể loại bỏ load và chỉ sử dụng authorize.

- `load_and_authorize_resource` : Thực thi before_action .Sẽ sử dụng kết hợp `load_resource` và `authorize_resource` , load resource có công dụng mặc định sẽ load các instance variable theo params của các action Rails. Trong trường hợp này nếu em vẫn sử dụng cách cũ cũng sẽ không load 2 lần, vì nó sẽ không load lại khi instance variable đã được gán.`

- authorize_resource` : Thực thi before_action .Sẽ cung cấp tự động xác thực tất cả các action trong controller.